lwcell/src/apps/mqtt/lwcell_mqtt_client.c

```diff
@@ -113,6 +113,7 @@ typedef enum {
 #define MQTT_RCV_GET_PACKET_TYPE(d)     ((mqtt_msg_type_t)(((d) >> 0x04) & 0x0F))
 #define MQTT_RCV_GET_PACKET_QOS(d)      ((lwcell_mqtt_qos_t)(((d) >> 0x01) & 0x03))
 #define MQTT_RCV_GET_PACKET_DUP(d)      (((d) >> 0x03) & 0x01)
+#define MQTT_RCV_GET_PACKET_RETAIN(d)   ((d & 0x01))
```

Comment:
Please put d in the brackets. `((d) & 0x01)`

Reply:
done.

Reply:
@MaJerle Hi, is there anything left to do?